src/components/NFTPage/Recommendation.vue

```diff
@@ -88,60 +94,139 @@ import { Swiper, SwiperSlide } from 'vue-awesome-swiper';
 import { DEFAULT_RECOMMENDATIONS_LIST } from '~/constant';
 
 import bookstoreMixin from '~/mixins/bookstore';
+import nftOrCollection from '~/mixins/nft-or-collection';
+import walletMixin from '~/mixins/wallet';
+import { mapActions, mapGetters } from 'vuex';
+
+const DISPLAY_ITEM_COUNT = 5;
 
 export default {
   name: 'NFTPageRecommendation',
   components: {
     Swiper,
     SwiperSlide,
   },
-  mixins: [bookstoreMixin],
+  mixins: [bookstoreMixin, nftOrCollection, walletMixin],
   props: {
-    iscnOwner: {
+    classId: {
+      type: String,
+      default: '',
+    },
+    collectionId: {
       type: String,
-      default: undefined,
+      default: '',
     },
-    isFollowed: {
+    // if true, show author works, else show default featured works
+    isAuthorSpecific: {
```

Comment:
Should this be a computed value that we have `(!classId && ! collectionId)` ? Or even just judge with `this.productOwner` exists and don't use a new flag

Reply:
[♻️ Simplify logic for determining default display](https://github.com/likecoin/liker-land/pull/1863/commits/a66dbc9fbf89afb0dc08753c0774612b6dc4ba0c)